csharp/src/Ice/IceDiscovery/Lookup.cs

```diff
@@ -275,41 +250,48 @@ internal class Lookup : ILookup
         }
     }
 
-    internal class LookupReply : ILookupReply
+     // TODO: missing IDisposable?
```

Comment:
Yes, we should indeed call dispose on the cancellation token source.

Reply:
For a follow-up PR. I leave the TODO for now.